src/pudl_archiver/archivers/classes.py

```diff
@@ -165,6 +164,42 @@ async def get_hyperlinks(
 
         return hyperlinks
 
+    def validate_archive(
+        self,
+        baseline_datapackage: DataPackage | None,
+        new_datapackage: DataPackage,
+        resources: dict[str, ResourceInfo],
+    ) -> bool:
+        """Run a series of validation tests for a new archive, and return results.
+
+        Args:
+            baseline_datapackage: DataPackage descriptor from previous version of archive.
+            new_datapackage: DataPackage descriptor from newly generated archive.
+            resources: Dictionary mapping resource name to ResourceInfo.
+
+        Returns:
+            Bool indicating whether or not all tests passed.
+        """
+        validation_tests = self.dataset_validate_archive(
+            baseline_datapackage, new_datapackage, resources
+        )
+        test_results = reduce(
+            lambda result, test: result and (test.result or test.ignore_failure),
+            validation_tests,
+            True,
+        )
+
+        return test_results
+
+    def dataset_validate_archive(
+        self,
+        baseline_datapackage: DataPackage | None,
+        new_datapackage: DataPackage,
+        resources: dict[str, ResourceInfo],
+    ) -> list[ValidationTestResult]:
+        """Hook to add archive validation tests specific to each dataset."""
+        return []
```

Comment:
I think we need at least one example validation test that we can use as a template to understand what these look like. Maybe a test that checks none of the files which were present in the previous archive have disappeared from the new archive?  What do we put here? The actual tests inline in this function? Or do we define other methods that are specific to the child class which pertains to a particular dataset, and then compile a list of their results here and return it?

Reply:
I added an example that checks for missing files from the previous archive. I think the cleanest way to add new tests would be to define a method for each test, and then call each method and compile results in a list. This is more or less what I've done in the example I added.